blaze-server/src/main/scala/org/http4s/server/blaze/Http1ServerStage.scala

```diff
@@ -189,20 +192,24 @@ private[blaze] class Http1ServerStage[F[_]](
       case Right(req) =>
         executionContext.execute(new Runnable {
           def run(): Unit = {
-            val action = Sync[F]
-              .suspend(raceTimeout(req))
+            val action = F
+              .defer(raceTimeout(req))
               .recoverWith(serviceErrorHandler(req))
               .flatMap(resp => F.delay(renderResponse(req, resp, cleanup)))
 
             parser.synchronized {
-              cancelToken = Some(
-                F.runCancelable(action) {
-                  case Right(()) => IO.unit
-                  case Left(t) =>
-                    IO(logger.error(t)(s"Error running request: $req")).attempt *> IO(
-                      closeConnection())
-                }.unsafeRunSync())
+              // TODO: review blocking compared to CE2
```

Comment:
I'm having a hard time digesting the synchronization here when it runs a background job.  I don't think you changed the semantics, but this is ... odd looking.

Reply:
The next revision will make it a lot more clearer, but it basically synchronizes on starting the effect and retrieving the cancel token, which is protected by `parser`